src/func/Microsoft.Health.Fhir.Ingest.Host/Startup.cs

```diff
@@ -0,0 +1,62 @@
+﻿// -------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+using System;
+using System.Linq;
+using EnsureThat;
+using Microsoft.ApplicationInsights;
+using Microsoft.ApplicationInsights.Extensibility;
+using Microsoft.Azure.Functions.Extensions.DependencyInjection;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Health.Fhir.Ingest.Telemetry;
+
+[assembly: FunctionsStartup(typeof(Microsoft.Health.Fhir.Ingest.Service.Startup))]
+
+namespace Microsoft.Health.Fhir.Ingest.Service
+{
+    public class Startup : FunctionsStartup
+    {
+        public override void Configure(IFunctionsHostBuilder builder)
+        {
+            EnsureArg.IsNotNull(builder, nameof(builder));
+
+            var instrumentationKey = Environment.GetEnvironmentVariable("APPINSIGHTS_INSTRUMENTATIONKEY");
+
+            if (instrumentationKey != null)
+            {
+                var configDescriptor = builder.Services.SingleOrDefault(tc => tc.ServiceType == typeof(TelemetryConfiguration));
+                if (configDescriptor?.ImplementationFactory == null)
+                {
+                    return;
```

Comment:
I think this might be cleaner if you do something like this:

builder.Services.BuildServiceProvider();
TelemetryConfiguration config = serviceProvider.GetService<TelemetryConfiguration>();

We do something similar in MeasurementFhirImportExtensions.AddMeasurementFhirImport

Reply:
I attempted to do this, and timeboxed the work at around 2 hours, but wasn't able to get this working by getting the service by type using serviceProvider.GetService<TelemetryConfiguration>() or typeof(TelemetryConfiguration) so am going to leave this as is for now